### PR TITLE
Several updates of sheet2code-ui

### PR DESCRIPTION
**Pull request by YuCJ**
Main updates:

- Fix uncatched error
- Add new props: `previewAllowToggleDisplay` and `previewDefaultDisplay` (For `scrollable-video`. It needs to measure the elements' sizes on mounting. So the elements cannot be hidden on mounting.)
- Fix DOM validation warning of nested p element

### CODE REVIEW

File: packages/sheet2code-ui/README.md [hunk 1]
```diff
@@ -32,21 +32,44 @@ ReactDOM.render(<BuildCodeUI.Component />, document.getElementById('root'))
 #### Props
 
 ```js
+/**
+ *
+ *
+ * @export
+ * @param {*} props
```
Comment (nickhsine) [inline packages/sheet2code-ui/README.md:39]: `@param {Object} props` ?

File: packages/sheet2code-ui/README.md [hunk 2]
```diff
@@ -32,21 +32,44 @@ ReactDOM.render(<BuildCodeUI.Component />, document.getElementById('root'))
...
+ * @param {string} props.codeLabel -The label of form text field for result code
+ * @param {string} props.codePathInAxiosResponse - The path to the returned code string in axios response
+ * @param {string[]} props.description - The description of the form
+ * @param {Function} props.errorToClientMessage - The function that take axios response error and give client error message
+ * @param {Function} props.formValuesToRequestConfig - The function that takes form values and returns axios request config
+ * @param {Function} props.getCodeFromAxiosResponse - The function that retrieves code string from axios response
+ * @param {number|'dynamic'} props.nOfSheetFields - Set how many sheet fields showed. 'dynamic' will showed at least one field for sheet.
```
Comment (nickhsine) [inline packages/sheet2code-ui/README.md:46]: `@param {number|string} props.nOfSheetFields` ?

File: packages/sheet2code-ui/src/components/app.js [hunk 3]
```diff
@@ -52,6 +52,24 @@ function codeReducer(state, action) {
   }
 }
 
+/**
+ *
+ *
+ * @export
+ * @param {*} props
```
[the remark this answers is not in the document]
Comment (YuCJ) [reply packages/sheet2code-ui/src/components/app.js:59]: It corresponds to the prop type:

```js
App.propTypes = {
  /* ... */
  nOfSheetFields: PropTypes.oneOfType([
      PropTypes.oneOf(['dynamic']),
      PropTypes.number,
  ]),
}
```

Quoted strings are supported by jsdoc:
https://stackoverflow.com/a/34744819/5814542

File: packages/sheet2code-ui/src/components/app.js [hunk 4]
```diff
@@ -52,6 +52,24 @@ function codeReducer(state, action) {
...
+ * @param {string} props.codeLabel -The label of form text field for result code
+ * @param {string} props.codePathInAxiosResponse - The path to the returned code string in axios response
+ * @param {string[]} props.description - The description of the form
+ * @param {Function} props.errorToClientMessage - The function that take axios response error and give client error message
+ * @param {Function} props.formValuesToRequestConfig - The function that takes form values and returns axios request config
+ * @param {Function} props.getCodeFromAxiosResponse - The function that retrieves code string from axios response
+ * @param {number|'dynamic'} props.nOfSheetFields - Set how many sheet fields showed. 'dynamic' will showed at least one field for sheet.
```
[the remark this answers is not in the document]
Comment (YuCJ) [reply packages/sheet2code-ui/src/components/app.js:66]: It corresponds to the prop type:

```js
App.propTypes = {
  /* ... */
  nOfSheetFields: PropTypes.oneOfType([
      PropTypes.oneOf(['dynamic']),
      PropTypes.number,
  ]),
}
```

Quoted strings are supported by jsdoc:
https://stackoverflow.com/a/34744819/5814542

[the remark this answers is not in the document]
Comment (nickhsine) [reply packages/sheet2code-ui/src/components/app.js:66]: Screen Shot 2020-07-27 at 6 05 10 PM
Not sure whether the result is what we want or not.
I use `jsdoc` to generate a dummy function, whose `@param {number|'dynamic'}`.
And the result shows that type could be `number|'dynamic'`, however, 'dynamic' is not a type tho.

[the remark this answers is not in the document]
Comment (YuCJ) [reply packages/sheet2code-ui/src/components/app.js:66]: Strings like `'dynamic'` are accepted as string literal types in jsdoc now, but they has not been added to the official document of jsdoc yet. See: https://github.com/jsdoc/jsdoc.github.io/issues/153  https://github.com/jsdoc/jsdoc/issues/629#issuecomment-58766768

The string literal types are also be used in TypeScript:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types

**Comment by nickhsine** [comment]
LGTM 👍

**Pull request merged**